api/ModelHandler.py

```diff
@@ -0,0 +1,314 @@
+import os
+import PeakError
+import threading
+import api.plLocks as locks
+import pandas as pd
+import api.TrackHandler as th
+import api.PLConfig as pl
+import api.JobHandler as jh
+
+summaryColumns = ['regions', 'fp', 'possible_fp', 'fn', 'possible_fn', 'errors', 'penalty', 'numPeaks']
+modelColumns = ['chrom', 'chromStart', 'chromEnd', 'annotation', 'height']
+
+
+def getModel(data):
+    data['hub'] = data['name'].split('/')[0]
+
+    data['track'] = data['name'].split('/')[-1]
+
+    problems = th.getProblems(data)
+
+    output = []
+
+    lock = locks.getLock(data['name'])
+
+    lock.acquire()
+
+    for problem in problems:
+        # data path, hub path, track path, problem path
+        modelsPath = '%s%s/%s/%s-%s-%s/' % (pl.dataPath, data['hub'], data['track'],
+                                            problem['ref'], problem['start'], problem['end'])
+
+        summaryPath = '%smodelSummary.txt' % modelsPath
+
+        # Converter required because pandas will lose trailing 0's
+        try:
+            summary = pd.read_csv(summaryPath, sep='\t', header=None, converters={6: lambda x: str(x)})
+        except FileNotFoundError:
+            continue
+
+        summary.columns = summaryColumns
+
+        minError = summary[summary['errors'] == summary['errors'].min()]
+
+        # Uses first penalty with min label error
+        # This will favor the model with the lowest penalty, given that summary is sorted
+        penalty = minError['penalty'].iloc[0]
+
+        minErrorModelPath = '%s%s_Model.bedGraph' % (modelsPath, penalty)
+
+        # TODO: If no good model, do LOPART
+
+        model = loadModelFile(minErrorModelPath, data)
+
+        output.extend(model)
+
+    lock.release()
+
+    return output
+
+
+def loadModelFile(path, data):
+    refseq = data['ref']
+    start = data['start']
+    end = data['end']
+
+    output = []
+
+    with open(path) as model:
+        modelLine = model.readline()
+
+        while not modelLine == '':
+            lineVals = modelLine.split()
+            lineStart = int(lineVals[1])
+            lineEnd = int(lineVals[2])
+            lineLabel = lineVals[3]
```

Comment:
chrom, start, end, label = modelLine.split()

Reply:
even better, why not just use a csv reader? (pandas/numpy?)

Reply:
This makes sense, relooking at it, I know how I can do it better already